backend/src/apiserver/resource/resource_manager.go

```
@@ -562,6 +647,38 @@ func (r *ResourceManager) CreateRun(ctx context.Context, run *model.Run) (*model
 	}
 	executionSpec.SetExecutionNamespace(k8sNamespace)
 
+	// Extract parallelism from template and configure synchronization semaphores
+	if parallelism, ok := extractPipelineRunParallelism(tmpl); ok && parallelism > 0 && run.PipelineVersionId != "" {
+		// Create/update ConfigMap with semaphore limit
+		if err := r.upsertPipelineParallelismConfigMap(
+			ctx, k8sNamespace, run.PipelineVersionId, parallelism); err != nil {
+			if apierrors.IsForbidden(errors.Cause(err)) {
+				glog.Warningf("Skipping pipeline_run_parallelism ConfigMap update in namespace %q: %v. Semaphore will not be configured.", k8sNamespace, err)
+				// Don't configure semaphores if ConfigMap creation fails
```

I think we should return an error.